### PR TITLE
Architecture

### PR DESCRIPTION
**Pull request by jpb06**
Several subjects dogged here:
- Bumping deps
- Frontend architecture overall
- Adding a backend route to fetch user profile (so that we can demonstrate user home page in storybook)
- Storybook: move from a "one storybook by app/lib" to only one storybook strategy. This will make it easier to find components in storybook and it will also be easier to build/deploy for cd.
- Intercepting XHR calls in storybook using msw
- Api layer refac: defining msw handlers in the api layer 
- Cypress: launching both backend and frontend locally
- Assets integration: components shared by multiple apps may have assets that must be integrated to the apps using them. The assets are thus defined in libraries and copied to the apps on build 
- A bunch of documentation changes: 
  - apps/libs readmes
  - nx
  - architecture
  - testing strategy

### CODE REVIEW

File: apps/front/src/templates/signup-form/hooks/useSignupFormSchema.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export type FormModel = {
   idSkills: number[];
 };
 
-export const useFormSchema = () => {
+export const useSignupFormSchema = () => {
```
Comment (balzdur) [inline apps/front/src/templates/signup-form/hooks/useSignupFormSchema.ts:16]: As a reminder, we would need to change our convention to pass translated messages directly (and not the translation key)

The change is still waiting to be done on `hublo-front` also by the way

**Review by balzdur — COMMENTED** [review]
Great job 👍 
The core changes are on the doc (huge, huge work on it) but since it's unreadable in PR diff inspector, I prefer you to merge it, I pull, I read, I make a PR if there is anything I want to change/precise or whatever (sounds easier to iterate over this way for doc stuff)

File: apps/front/public/mockServiceWorker.js [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+
+const INTEGRITY_CHECKSUM = '02f4ad4a2797f85668baf196e553d929'
+const bypassHeaderName = 'x-msw-bypass'
+const activeClientIds = new Set()
+
+self.addEventListener('install', function () {
+  return self.skipWaiting()
+})
+
+self.addEventListener('activate', async function (event) {
```
[the remark this answers is not in the document]
Comment (jpb06) [reply apps/front/public/mockServiceWorker.js:19]: No this is autogenerated: 
https://mswjs.io/docs/cli/init

For the docs, you can access them directly on the branch:
https://github.com/jpb06/mui-rhf-sandbox/blob/architecture/README.md

**Comment by jpb06** [comment]
Thanks for your feedback on this @balzdur !

About the docs, you can access them directly on the branch:
https://github.com/jpb06/mui-rhf-sandbox/blob/architecture/README.md

File: apps/front/src/templates/signup-form/SignupForm.spec.tsx [hunk 3]
```diff
@@ -376,17 +375,22 @@ describe('Signup component', () => {
 
       expect(callCount).toBe(2);
       server.events.removeListener('request:match', cb);
-    });
+    }, 60000);
```
[the remark this answers is not in the document]
Comment (jpb06) [reply apps/front/src/templates/signup-form/SignupForm.spec.tsx:380]: You're right, it's a bit overkill! 

The story behind it is we found out with Jon nx was kind of restrictive with tests duration, so we had tests failing with `nx affected`.

File: docs/react-hook-form.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
-## ⚡ Forms handling
+# ⚡ Forms handling
```
Comment (ArthurGoupil) [inline docs/react-hook-form.md:2]: Maybe we should update this with latest chapter decision:
- only use translated error messages in schema
- use a custom resolver that handle defaults error messages (already implemented in hublo-front)

File: libs/front/api/src/main-backend/calls/useUserDataQuery.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+//import useLocalStorageState from 'use-local-storage-state';
```
Comment (ArthurGoupil) [inline libs/front/api/src/main-backend/calls/useUserDataQuery.ts:1]: to remove

File: libs/front/api/src/main-backend/calls/useUserDataQuery.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    undefined;
+  }
+
+  return JSON.parse(storedToken as string);
+};
+
+export const useUserDataQuery = () => {
+  // const [token] = useLocalStorageState<string>('token', {
+  //   ssr: true,
+  // });
```
Comment (ArthurGoupil) [inline libs/front/api/src/main-backend/calls/useUserDataQuery.ts:25]: to remove

File: libs/front/components/src/molecules/layout/fullpage-box/FullpageBox.tsx [hunk 7]
```diff
@@ -19,7 +19,7 @@ export const FullpageBox: React.FC = ({ children }) => (
     justifyContent="center"
     sx={{
       minHeight: '100vh',
-      // backgroundImage: 'url("/img/dots.gif")',
+      //backgroundImage: 'url("/img/dots.gif")',
```
Comment (ArthurGoupil) [inline libs/front/components/src/molecules/layout/fullpage-box/FullpageBox.tsx:22]: to remove ?

File: apps/front/src/templates/logged-user-home/LoggedUserHome.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    <>
+      {
+        {
+          idle: <FullpageLoader />,
+          redirecting: <FullpageLoader />,
+          loading: <FullpageLoader />,
+          error: <UserProfileLoadingError />,
+          success: <UserProfile user={user as User} />,
+        }[status]
+      }
```
[the remark this answers is not in the document]
Comment (jpb06) [reply apps/front/src/templates/logged-user-home/LoggedUserHome.tsx:21]: Yes. This is something I personnaly disagree with :)

File: apps/front/src/templates/logged-user-home/molecules/user-profile-loading-error/UserProfileLoadingError.tsx [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import { FullpageError } from '../../../../molecules';
```
[the remark this answers is not in the document]
Comment (jpb06) [reply apps/front/src/templates/logged-user-home/molecules/user-profile-loading-error/UserProfileLoadingError.tsx:1]: Path aliases can only be set for apps/libs, not within apps/libs

File: apps/front/src/templates/logged-user-home/LoggedUserHome.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+import { msw } from '@api/msw';
+import { ReactQueryWrapper } from '@tests';
+import { mockedUser } from '@tests/mocked-data';
+
+import { LoggedUserHome } from './LoggedUserHome';
+
+export default {
+  component: LoggedUserHome,
+  title: 'Front app/User stories/Logger user',
```
[the remark this answers is not in the document]
Comment (jpb06) [reply apps/front/src/templates/logged-user-home/LoggedUserHome.stories.tsx:11]: I don't believe we should no. The idea here is to make it easy as well to navigate and find components for people outside the dev team

So we would have
- Shared 
  + molecules
  | Component A
  | Component B
  | ...
  + organisms
  | Component C
  | ...
- Front App 1
  + molecules
  | Component D
  | ...
  + organisms
  | Component E
  | ...
  + User stories
  | User story 1
  | User story 2
  | ...

File: apps/front/src/templates/logged-user-home/LoggedUserHome.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+} as ComponentMeta<typeof LoggedUserHome>;
+
+const Template: Story = (_) => (
+  <ReactQueryWrapper>
+    <LoggedUserHome />
+  </ReactQueryWrapper>
+);
+
+export const NominalCase = Template.bind({});
+NominalCase.args = {};
```
[the remark this answers is not in the document]
Comment (jpb06) [reply apps/front/src/templates/logged-user-home/LoggedUserHome.stories.tsx:21]: I prefer to keep these for now; work on storybook is only half done anyway so I will get back to this!

File: docs/react-hook-form.md (hunk 4 above)
Comment (jpb06) [reply docs/react-hook-form.md:2]: I will update the docs once I'm done with translations (will be in another PR)

File: apps/front/src/templates/logged-user-home/molecules/user-avatar/UserAvatar.tsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+        {...stringAvatar(`${firstName} ${lastName}`)}
+        sx={{
+          width: 80,
+          height: 80,
+          mb: 2,
+          fontSize: 'xx-large',
+        }}
+      />
+    </Grid>
+  </Grid>
```
[the remark this answers is not in the document]
Comment (jpb06) [reply apps/front/src/templates/logged-user-home/molecules/user-avatar/UserAvatar.tsx:30]: I know this is something frowned upon by @PupoSDC for example, but I personally love sx.

We began discussing about it [here](https://hubloteam.slack.com/archives/C0156ND868M/p1645014419011739?thread_ts=1645006084.836269&cid=C0156ND868M).
Perhaps we should do a chapter meeting for this?

**Review by ArthurGoupil — APPROVED** [review]

**Comment by jpb06** [comment]
> https://github.com/jpb06/mui-rhf-sandbox/pull/16#discussion_r822504335

It will never happen since we are in the `success case`

**Pull request merged**